### ISSUE TITLE
InputText() old inputs change simultaneously with new user inputs

### ISSUE THREAD

**Issue by svkvarun43** [ocornut/imgui#5383]
**Version/Branch of Dear ImGui:**

Version: 1.88
Branch: master

**My Issue/Question:**

I am getting three user input through `ImGui::InputTextWithHint() ` , The inputs get display on screen, When i edit the user input the previous input displayed on the screen also changes.

**Screenshots/Video**
My text input window, which gets input from user

![image](https://user-images.githubusercontent.com/98422353/172896527-5721b097-3c7e-4dd2-beb3-7b6ec5add962.png)

It gets displayed 

![image](https://user-images.githubusercontent.com/98422353/172896768-4ea1b9da-21d3-4a73-9026-bbd8539ae2ad.png)

When I redit my user input, the text change simultaneously in the display area too, How can I avoid this happening? 

![image](https://user-images.githubusercontent.com/98422353/172897320-29645ec2-e735-476f-9525-d39cb3e2e3f0.png)

I am looking to display the new user input than retaining the previous entered input

**Snippet which gets the user input:**

```
ImGui::InputTextWithHint("Host Machine", "<localhost>", r.host_machine, IM_ARRAYSIZE(r.host_machine));
ImGui::InputTextWithHint("Port", "<9090>", r.port, IM_ARRAYSIZE(r.port), ImGuiInputTextFlags_CharsDecimal);
ImGui::InputTextWithHint("Topics", "</Oct5Boom>", r.topics, IM_ARRAYSIZE(r.topics));
```

**Comment by ocornut** [comment]
Use two sets of variables..

**Comment by ocornut** [comment]
> Use two sets of variables..
You are currently unhappy that a variable you give to InputText() is edited, that its purpose. If you use a single set of variable there's no concept of "previous input" and "new user input" here.

Closing as answered but feel free to comment if you are unsure how to proceed.

**Comment by svkvarun43** [comment]
True, I am a bit stuck on how to proceed unfortunately, could you help me please. Maybe a small example how to avoid the simultaneous change please.


For ref,
My goal was to get user inputs into the textinput and print them on screen on clicking "OK"

Once it has been passed and display, the field should be cleared for getting new user textinput.
